### ISSUE TITLE
List of possible things to work on

### ISSUE THREAD

**Issue by fvictorio** [poanetwork/ico-wizard#422]
It'd be great to have more issues in the backlog so that @fernandomg doesn't get out of things to do while I'm on vacation the next two weeks. Here's a list of suggestions, let me know if you agree with them and I'll fill an issue for each one.

## Bugs

1. Sometimes gas price is set to 0 when trying to contribute in the invest page.
1. In step two, validate that the number of decimals is between 0 and 18.

## Usability

1. Change "Disable whitelisting" (defaulted to Yes) to "Enable whitelisting" (defaulted to No).
1. When reserving tokens, if the same address is input twice, replace the previous occurrence in the list.
1. Same as 2, but for whitelist.

## Refactoring

1. Make RadioInputField more generic (right now it can only have two options).

I'll edit the list if I can think of something else.

**Comment by igorbarinov** [comment]
integration with gas price oracle is very important

**Comment by fvictorio** [comment]
Yes, I didn't include issues that are already in submitted. We'll make that top priority, though.

**Comment by pablofullana** [comment]
Added a new entry to *Usability* list:

_As a user, when creating a crowdsale, it would be nice to see the gas price for the current network instead of always main one._

**Comment by vbaranov** [comment]
Added new entry to **Usability** list:

*If user forgot to add reserved tokens list at deployment stage, give this opportunity to him on managing page*

**Comment by TRIHRD** [comment]
I need to retain 50% tokens for future offline wallet storage and necessary to be able to swap / Tethered for fiat currency for working capital to  NZD- AUD?. 100 million total capx. coins . First tier 5 for 1 ,
2nd tier  2 for 1 ETH, not wanting white paper , keeping the raising small with big ideas.  thanks people.

**Comment by TRIHRD** [comment]
Was first saving 80% in reserves  so if crowd sale makes 20% i would be happy ,But thought why not open it up and see holding 50% in reserves as long as it is all very secure safe.

**Comment by vbaranov** [comment]
Added new entry to **Usability** list:

*Add explanation or error if contribution transaction failed in `./invest` page*

**Comment by vbaranov** [comment]
Added new entry to **Usability** list:

*Consider download contract info zip file from /manage page.*

**Comment by pablofullana** [comment]
Added 2 new entries to *Usability* list:

_(Feature) 'With funding Goal' strategy for crowdsale contracts (#613)_
_(Feature) Widget for crowdsale stats (#615)_

**Comment by pablofullana** [comment]
Added 2 new entries to **Usability** list:

_(Feature) Stasts section (#620)_

**Comment by pablofullana** [comment]
Added 2 new entries to **Usability** list:

_(Feature) Set tier to start automatically once previous one is over (#624)_
_(Feature) Goal for crowdsale tiers (#625)

**Comment by pablofullana** [comment]
Added a new entry to Usability list:

_As I user, when setting up a new crowdsale, I'd like to be able to associate an image to the new Token (#629)_

**Comment by pablofullana** [comment]
Added a new entry to **Usability** list:

_(Feature) Discount option: Timeframe, first n tokens, promo code (#649)_
